clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -404,6 +411,12 @@ private void maybeTriggerWakeup() {
             throw new WakeupException();
         }
     }
+    
+    private void maybeThrowInterruptException() {
```

Comment:
"Usually" was probably a poor choice of words. Users sometimes use `wakeup()` for other purposes than shutdown (e.g. to change subscriptions), in which case, an additional flag is checked to determine whether to keep going in the loop or to shutdown. The main point is that the consumer may already be in `close()` when the user calls `wakeup()`, so we opted to ignore any `wakeup()` received in `close(). That said, this kind of usage is a little uncommon, so it wouldn't hurt to change the example in the javadoc to a pattern relying only on `wakeup()` for shutdown.

I'm leaning toward letting the interrupt semantics differ from wakeup (as implemented in the current patch) as long as we're clear in the docs that `wakeup()` is the preferred (i.e. the safest) option. Just one point of clarification: the only time we disable wakeups in a blocking operation is in `close()`, and this only really matters if auto-commit is enabled. For a direct call to the `commitSync()` API, using `wakeup()` should work as intended. So if we wanted to mimic "wakeup semantics" for interrupts, that would be the only limitation.